bluetooth/src/iosMain/kotlin/BluetoothMonitor.kt

```diff
@@ -17,50 +17,49 @@
 
 package com.splendo.kaluga.bluetooth
 
+import com.splendo.kaluga.base.ServiceMonitor
 import com.splendo.kaluga.logging.debug
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.asStateFlow
 import platform.CoreBluetooth.CBCentralManager
 import platform.CoreBluetooth.CBCentralManagerDelegateProtocol
 import platform.CoreBluetooth.CBCentralManagerStatePoweredOn
 import platform.darwin.NSObject
 
 actual class BluetoothMonitor internal constructor(
     private val centralManager: CBCentralManager
-) {
+) : ServiceMonitor() {
 
     actual class Builder actual constructor() {
         actual fun create() = BluetoothMonitor(centralManager = CBCentralManager())
     }
 
-    private val centralManagerDelegate = object : NSObject(), CBCentralManagerDelegateProtocol {
+    internal class CentralManagerDelegate(
+        private val updateEnabledState: () -> Unit
+    ) : NSObject(), CBCentralManagerDelegateProtocol {
         override fun centralManagerDidUpdateState(central: CBCentralManager) {
             updateEnabledState()
```

Comment:
Same here, call function from abstract class

Reply:
I can call `updateState()` from start/stopMonitoning methods, but this is the class that receives the updates and I have no access to it from base class

Reply:
Ah, I see, then you can pass updateState to constructor here CentralManagerDelegate(::updateState)?